content/docs/lists-and-keys.md

```diff
@@ -119,34 +123,35 @@ const todoItems = todos.map((todo) =>
 );
 ```
 
-When you don't have stable IDs for rendered items, you may use the item index as a key as a last resort:
+Ketika anda tidak memiliki ID yang stabil untuk me-*render* item, anda bisa menggunakan index dari item sebagai *key* sebagai *key* untuk pilihan terakhir:
 
 ```js{2,3}
 const todoItems = todos.map((todo, index) =>
   // Only do this if items have no stable IDs
+  // Hanya lakukan ini jika item tidak memiliki ID yang stabil
   <li key={index}>
     {todo.text}
   </li>
 );
 ```
 
-We don't recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state. Check out Robin Pokorny's article for an [in-depth explanation on the negative impacts of using an index as a key](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). If you choose not to assign an explicit key to list items then React will default to using indexes as keys.
+Kami tidak merekomendasikan menggunakan indeks untuk *key* jika urutan item nantinya berubah. Ini berdampak negatif terhadap kinerja dan dapat menyebabkan masalah dengan state komponen. Simak artikel Robin Pokorny untuk [penjelasan lebih dalam mengenai dampak negatif penggunaan index sebagai *key*](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). Jika anda memilih untuk tidak menetapkan *key* pada *list* item maka React secara bawaan akan menggunakan indeks sebagai *key*.
 
-Here is an [in-depth explanation about why keys are necessary](/docs/reconciliation.html#recursing-on-children) if you're interested in learning more.
+Berikut adalah [penjelasan lebih dalam tentang kenapa *key* sangat diperlukan](/docs/reconciliation.html#recursing-on-children) Jika anda tertarik untuk belajar lebih lanjut.
 
-### Extracting Components with Keys {#extracting-components-with-keys}
+### Mengekstrak Komponen dengan *Key* {#extracting-components-with-keys}
 
-Keys only make sense in the context of the surrounding array.
+*Key* hanya akan di gunakan di konteks array yang mengurung item dengan *key* tersebut
```

Comment:
Bagian ini masih terasa janggal.

Reply:
*Key* hanya dapat digunakan dalam ruang lingkup *array* saja.

When do we know to use the singular form instead of plural? like *Key* vs *Keys*.

Reply:
https://github.com/reactjs/id.reactjs.org/wiki/Glosarium please refer here for the usage of loanwords. 

| English | Indonesian |
| ------ | ------ |
| array | senarai |